soci/index_builder.go

```diff
@@ -83,17 +86,23 @@ func WithBuildToolIdentifierOption(buildToolIdentifier string) IndexBuilderOptio
 	}
 }
 
-// BuildIndex builds SOCI index for an image
+func WithPlatformOption(platform ocispec.Platform) IndexBuilderOption {
+	return func(b *IndexBuilder) error {
+		b.platform = platform
+		return nil
+	}
+}
+
+// BuildIndex builds SOCI index for an image target (ocispec.Descriptor)
 // This returns SOCI index manifest and an array of io.Reader of zTOCs
-func (b *IndexBuilder) BuildIndex(ctx context.Context, desc ocispec.Descriptor) (*Index, []io.Reader, error) {
-	platform := platforms.Default()
+func (b *IndexBuilder) BuildIndex(ctx context.Context, desc ocispec.Descriptor) (*IndexWithMetadata, []io.Reader, error) {
```

Comment:
nit: `IndexBuilder.BuildIndex` feels redundant. Can this just be called `Build`?

Reply:
yes